php/class-wp-app-manifest.php

```diff
@@ -0,0 +1,202 @@
+<?php
+/**
+ * WP_APP_Manifest class.
+ *
+ * @package PWA
+ */
+
+/**
+ * WP_APP_Manifest class.
+ *
+ * Mainly copied from Jetpack_PWA_Manifest and Jetpack_PWA_Helpers.
+ */
+class WP_APP_Manifest {
+
+	/**
+	 * The theme color to use if no dynamic value is present.
+	 *
+	 * @var string
+	 */
+	const FALLBACK_THEME_COLOR = '#fff';
+
+	/**
+	 * The REST API namespace for the manifest request.
+	 *
+	 * @var string
+	 */
+	const REST_NAMESPACE = 'wp/v2';
+
+	/**
+	 * The REST API route for the manifest request.
+	 *
+	 * @var string
+	 */
+	const REST_ROUTE = '/pwa-manifest';
```

Comment:
I think `/webmanifest` is better since `.webmanifest` it is the file extension. Or rather, we can call it `/web-app-manifest` to be more accurate.